app/models/resource_action_workflow.rb

```diff
@@ -21,14 +21,19 @@ def initialize(values, requester, resource_action, options = {})
 
   Vmdb::Deprecation.deprecate_methods(self, :dialogs => :dialog)
 
-  def submit_request
+  def submit_request(data)
```

Comment:
Should we provide a default value `(data = {})` which would remove the need to pass empty hashes in service_template and the specs?

Also, `service_template and the specs` is not a good band name.  🎻 🥁